public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -372,164 +477,216 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
     return null;
   }
   return (
-    <RuleEditorSection
-      stepNo={2}
-      title={sectionTitle}
-      description={
-        <Stack direction="row" gap={0.5} alignItems="center">
-          <Text variant="bodySmall" color="secondary">
-            {helpLabel}
-          </Text>
-          <NeedHelpInfo
-            contentText={helpContent}
-            externalLink={helpLink}
-            linkText={'Read more on our documentation website'}
-            title={helpLabel}
-          />
-        </Stack>
-      }
-    >
-      {/* This is the cloud data source selector */}
-      {isDataSourceManagedRuleByType(type) && (
-        <CloudDataSourceSelector onChangeCloudDatasource={onChangeCloudDatasource} disabled={editingExistingRule} />
-      )}
-
-      {/* This is the PromQL Editor for recording rules */}
-      {isRecordingRuleType && dataSourceName && (
-        <Field error={errors.expression?.message} invalid={!!errors.expression?.message}>
-          <RecordingRuleEditor
-            dataSourceName={dataSourceName}
-            queries={queries}
-            runQueries={() => runQueriesPreview()}
-            onChangeQuery={onChangeRecordingRulesQueries}
-            panelData={queryPreviewData}
-          />
-        </Field>
-      )}
-
-      {/* This is the PromQL Editor for Cloud rules */}
-      {isCloudAlertRuleType && dataSourceName && (
-        <Stack direction="column">
+    <>
+      <RuleEditorSection
+        stepNo={2}
+        title={sectionTitle}
+        fullWidth={true}
+        description={
+          <Stack direction="row" gap={0.5} alignItems="center">
+            <Text variant="bodySmall" color="secondary">
+              {helpLabel}
+            </Text>
+            <NeedHelpInfo
+              contentText={helpContent}
+              externalLink={helpLink}
+              linkText={'Read more on our documentation website'}
+              title={helpLabel}
+            />
+          </Stack>
+        }
+        switchMode={
+          isGrafanaManagedRuleByType(type) && isSwitchModeEnabled
```

Comment:
nit: can we pull this into a separate variable so we don't have to inline it in the markup? :)